app/operations/subject_groups/create.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module SubjectGroups
+  class Create < Operation
+    string :project_id
+    string :uploader_id
+    array :subject_ids
+
+    def execute
+      group_subjects_in_order = find_subjects_in_order(subject_ids)
+      ensure_all_subjects_exist(subject_ids.size, group_subjects_in_order.size)
+
+      subject_group = build_subject_group(group_subjects_in_order)
+
+      # ensure this save is 'isolated' from any surrounding transaction
+      # https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html#module-ActiveRecord::Transactions::ClassMethods-label-Nested+transactions
+      SubjectGroup.transaction(requires_new: true) do
+        # save this subject_group and all the associated records (members, group_subject)
+        subject_group.save!
+        # track the subject_group on the placeholder subject metadata
+        # useful for the subject exports and downstream analysis
+        # to know how the group is comprised of what subjects
+        # possible use in online retirement as well - TBD
+        update_group_subject_metadata(subject_group)
+      end
+      subject_group
+    end
+
+    private
+
+    def ensure_all_subjects_exist(subject_ids_size, group_subjects_in_order_size)
+      return if subject_ids_size == group_subjects_in_order_size
+
+      raise Error, 'Number of found subjects does not match the size of param subject_ids'
+    end
+
+    def build_subject_group(subjects)
+      subject_group = SubjectGroup.new(project: project, key: subject_group_key)
+      subjects.each_with_index do |subject, index|
+        subject_group.members << build_subject_group_member(subject_group, subject, index)
+      end
+      subject_group.group_subject = build_group_subject
+      subject_group
+    end
+
+    def subject_group_key
+      subject_ids.join('-')
+    end
+
+    def project
+      @project ||= Project.find(project_id)
+    end
+
+    def uploader
+      @uploader ||= User.find(uploader_id)
+    end
+
+    def find_subjects_in_order(subject_ids)
+      @find_subjects_in_order ||=
```

Comment:
Should we also be ensuring that all subjects belong to the correct project?

Reply:
interesting - i don't think we should restrict a subject group this way as groups may be comprised of subjects that are re-used across projects (a feature we have in the API). 

I'd rather leave this to the selector systems / input to the group to handle and leave this check out of the group.